documentation/architecture.md

```diff
@@ -0,0 +1,247 @@
+# Clear Containers Architecture
```

Comment:
General feedback ... this is a **great** document - one we have had the fragments of scattered around here and there, and were quite over-due pulling it all together in one cohesive place - so, thanks for doing that.
Most comments below are just stylistic or typo nitpicks